client/packages/system/src/IndicatorsDemographics/DetailView/IndicatorsDemographics.tsx

```diff
@@ -75,54 +77,50 @@ const IndicatorsDemographicsComponent = () => {
   };
 
   const setter = (patch: RecordPatch<Row>) => {
-    const updatedDraft = { ...draft };
     const percentage = !patch.percentage ? 0 : patch.percentage;
     const percentageChange = percentage != draft[patch.id]?.percentage;
 
     setIsDirty(true);
 
+    const existingRow = draft[patch.id];
+    if (!existingRow) return;
+
+    const patchedRow: Row = { ...existingRow, ...patch };
+
     // change state of name only if only name changes
     if (!percentageChange) {
-      setDraft({ ...updatedDraft, [patch.id]: { ...patch } as Row });
+      setDraft({ ...draft, [patch.id]: patchedRow });
       return;
     }
 
     const updatedRow = calculateAcrossRow(
-      { ...patch } as Row,
+      patchedRow,
       headerDraft,
       indexPopulation
     );
-    setDraft({ ...updatedDraft, [patch.id]: updatedRow });
+    setDraft({ ...draft, [patch.id]: updatedRow });
   };
 
   // generic function for handling percentage change, and then re calculating the values of that year
-  const handleGrowthChange = (patch: RecordPatch<HeaderValue>) => {
+  const handleGrowthChange = (updatedHeader: HeaderData) => {
```

Comment:
Letting the `GrowthRow` (header data management component) manage the new shape of HeaderData, just updating here